clang/include/clang/CConv/ConstraintVariables.h

```diff
@@ -158,13 +158,29 @@ void constrainConsVarGeq(std::set<ConstraintVariable *> &LHS,
                       ConsAction CA,
                       bool doEqType,
                       ProgramInfo *Info);
+void constrainConsVarGeq(std::set<ConstraintVariable *> &LHS,
+                         std::set<ConstraintVariable *> &RHS,
+                         Constraints &CS,
+                         PersistentSourceLoc *PL,
+                         ConsAction CA,
+                         bool doEqType,
+                         bool derefLHS,
```

Comment:
I'd prefer that instead of a third and fourth `constrainConsVarGeq` that differs from the other two by only one parameter, you just add the parameter `derefLHS` to the existing prototypes, and then change the callers to make the parameter `false`. From a client API perspective, the two things are equivalent, so there should just be one way of doing it.

Reply:
So, basically: Drop the adapter code for the old prototype and change all the callers to use the new prototype.

Reply:
Fair enough. I avoided this to keep the diff smaller, but I think the IDE can just refactor this for me.